### DISCUSSION TITLE
how do you fix st.session_state has no key "GFPGAN_model"

### DISCUSSION THREAD

**Discussion by loki2020** [hlky/stable-diffusion-webui#1452] [Q&A] [answered]
I've got the webui.cmd (gradio) working great and now I have the webui-streamlit ui loading but when I hit generate I get no image and a big red box full of angry bees. any advice, I'm kind of slow and _i_ can't find any documented fix for this.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\loki\miniconda3\envs\ldm\lib\site-packages\streamlit\runtime\scriptrunner\script_runner.py", line 562, in _run_script
    exec(code, module.__dict__)
  File "C:\Users\loki\stable-diffusion-webui\scripts\webui_streamlit.py", line 177, in <module>
    layout()
  File "C:\Users\loki\stable-diffusion-webui\scripts\webui_streamlit.py", line 141, in layout
    layout()
  File "scripts\txt2img.py", line 406, in layout
    use_GFPGAN=st.session_state["use_GFPGAN"], GFPGAN_model=st.session_state["GFPGAN_model"] ,
  File "C:\Users\loki\miniconda3\envs\ldm\lib\site-packages\streamlit\runtime\state\session_state_proxy.py", line 93, in __getitem__
    return get_session_state()[key]
  File "C:\Users\loki\miniconda3\envs\ldm\lib\site-packages\streamlit\runtime\state\safe_session_state.py", line 111, in __getitem__
    return self._state[key]
  File "C:\Users\loki\miniconda3\envs\ldm\lib\site-packages\streamlit\runtime\state\session_state.py", line 440, in __getitem__
    raise KeyError(_missing_key_error_message(key))
KeyError: 'st.session_state has no key "GFPGAN_model". Did you forget to initialize it? More info: https://docs.streamlit.io/library/advanced-features/session-state#initialization'
![error](https://user-images.githubusercontent.com/1242975/194390524-d8dc54f2-b795-4377-bcc1-df2c3725ef62.jpg)

**Comment by codedealer** [comment]
try clearing userconfig in `configs/webui`

**Reply by loki2020** [reply]
Sadly, that solution didn't work. still getting he same error.

**Comment by loki2020** [comment, marked as answer]
so i was reading the userconfig file and I see 
GFPGAN_dir: ./models/gfpgan
  GFPGAN_model: GFPGANv1.4
I look in said folder and it contained GFPGANv1.3 so i added a copy of 1.4 and there ya go, working.

**Reply by TornSky** [reply]
Jello, Could you explain in a little more detail what you did exactly? I am having the same problem, I really don't understand what you did. I looked in:

stable-diffusion-webui/configs/webui
and opened the file called "userconfig_streamlit.yaml" but it already shows "GFPGAN_model: GFPGANv1.4" on my end.


please help!

**Reply by loki2020** [reply]
i opened the folder it is referencing. Stable-diffusion>models>gfpgan
there was one file, GFPGANv1.3.pth and that was not the file referenced in userconfig, so I added a copy of GFPGANv1.4.pth to the folder as well.

**Reply by TornSky** [reply]
okay I made a copy of "GFPGANv1.3.pth" and renamed it to "GFPGANv1.4.pth"

that still didn't work. then I tried restarting my computer. still nothing.

sorry I know I'm an idiot for not understanding this could you explain it to me  like it was a 5 year old lol
![Screenshot from 2022-10-06 19-20-02](https://user-images.githubusercontent.com/105568636/194453295-6066289d-e0da-453a-a93b-41a49270dbbc.png)

**Reply by TornSky** [reply]
Nm I figured it out.  I opened up streamlit and I Clicked on the"Model Manager" on the left side then I downloaded the "GFPGAN"

I didn't have a green check mark next to it. so I clicked "Download" then I let it do its thing. now it is working for me

Thanks!

**Comment by loki2020** [comment]
TY for the hint Codedealer.
